core/src/main/java/ai/timefold/solver/core/impl/score/director/stream/BavetConstraintStreamScoreDirector.java

```diff
@@ -52,9 +53,27 @@ public void clearShadowVariablesListenerQueue() {
         clearVariableListenerEvents();
     }
 
+    /**
+     * The function is exclusively available for the Bavet score director, and its use must be approached with caution.
+     * The primary purpose of this method is
+     * to enable the {@code ConstraintVerifier}
+     * to update the consistency status of entities in the solution without updating shadows.
+     * <p>
+     * This must be done before setWorkingSolutionWithoutUpdatingShadows, which inserts the entities.
+     */
+    public void updateConsistencyFromSolution(Solution_ solution) {
```

Comment:
I see the point of this for Bavet - but doesn't Incremental also have the same problem? It also loads the problem from a starting solution, and works with before/after calls.

Reply:
Although as I write this, I realize that `ConstraintVerifier` is, by definition, only applicable to CS.

Reply:
Yep; solves a similar but different issue as "clearShadowVariablesListenerQueue", where consistency needs to be updated as `ConstraintVerifier` bypasses the variable listener API.